### ISSUE TITLE
Edgeshark extcap not detected by Wireshark 4.4.0

### ISSUE THREAD

**Issue by pth2000** [siemens/cshargextcap#20]
I noticed that when I jump from edgeshark to Wireshark 4.4.0 and try to capture a packet, I get an error from Wireshark (I noticed Docker host capture and packetflix disappeared from Wireshark).

I think the Wireshark update may require further adaptation of the cshargextcap.

**Comment by pth2000** [comment]
Note: Use the Windows version

Similar to the problem: https://github.com/siemens/cshargextcap/issues/17

Thank you very much!

**Comment by thediveo** [comment]
Can you please provide more details?

**Comment by pth2000** [comment]
Thank you for your reply.

After updating 4.4.0, Docker host capture and packetflix disappeared from Wireshark:

![image](https://github.com/user-attachments/assets/96f375a1-9c9a-487e-8587-3a1548e54fa9)

Jump from edgeshark:

![image](https://github.com/user-attachments/assets/c6d98323-af80-4ec5-84c6-640b19b57c0f)

**Comment by thediveo** [comment]
You seem to be on Windows, can you please check that the cshargextcap.exe is still present in Wireshark's global extcap directory? This is to make sure that the Wireshark installer hasn't removed the plugin.

**Comment by cyb3n3tic** [comment]
Hi, I just ran into this myself - happened to update Wireshark on my Windows 11 machine and subsequently found that Edgeshark no longer worked. Reverting to older version of Wireshark has solved it for me for now.
In Wireshark 4.4.0 Manage interfaces listed as per attached if that helps? 
![Packetflix](https://github.com/user-attachments/assets/d0beb208-6ada-4d29-afd6-6e3bad66525d)

**Comment by pth2000** [comment]
I apologize for the late response. I reinstalled the older version of Wireshark, and csharkextcap is working fine, which might indicate that the plugin wasn't removed.

Regarding the extcap directory, cyb3rn3tic has already conducted checks on it.

I hope this helps.

**Comment by thediveo** [comment]
Thank you for verifying that the extcap wasn't removed during the Wireshark upgrade. My putonghua is very rusty, so I'm unclear a to what the error 123 message actually is. Can you please check with the Wireshark upstream what might break 3rd party extcaps?

**Comment by pth2000** [comment]
> Thank you for verifying that the extcap wasn't removed during the Wireshark upgrade. My putonghua is very rusty, so I'm unclear a to what the error 123 message actually is. Can you please check with the Wireshark upstream what might break 3rd party extcaps?

This error is:
```
[Error 123] The filename, directory name, or volume label syntax is incorrect.
```

In fact, I don't have a deep understanding of how Wireshark and 3rd party extcaps work, so I may not be able to provide very effective advice.

This is the upgrade document of Wireshark: https://www.wireshark.org/docs/relnotes/wireshark-4.4.0.html

I noticed that there are Major API Changes, including the changes of Plugins API.

I hope this helps.

**Comment by thediveo** [comment]
The release notes mention plugins which are completely different from extcaps. I haven't found any mention of extcap API changes.

**Comment by thediveo** [comment]
As an intermediate fix until we can release a new installer, please move the `cshargextcap-amd64.exe` binary from `Wireshark/extcap` into the folder `Wireshark/extcap/wireshark` (sic!) that should have been created by the Wireshark 4.4.0 installer. Can you please confirm that this works for you?

**Comment by thediveo** [comment]
We are still blocked by Wireshark upstream:
- https://gitlab.com/wireshark/wireshark/-/issues/20054: fixed, not yet released -- we unfortunately don't have the resources to test the commits on the range of platforms we are providing cshargextcap for. Normally, extension points should come with some kind of contract, but this seems to be different in case of Wireshark.
- https://gitlab.com/wireshark/wireshark/-/issues/20069: no response since last week

We don't have the resources to maintain different distro packages times different Wireshark versions. In addition, we're blocked by upstream breaking Wireshark upgrade paths for Edgeshark capture users. If Wireshark does not fix issue 20069, we're stuck as we don't have resources to potentially multiple times work around upstream breaking changes.

We would appreciate if new maintainers would step up willing to maintain multi-distro and multi-OS packaging and the associated documentation changes required. Please get in contact with the https://github.com/siemens/cshargextcap/blob/main/.github/CODEOWNERS if you are willing.

**Comment by vPackets** [comment]
Does this apply to all wireshark 4.4 version ? 

looks like same behavior on MacOS.... And we have no way to downgrade to an old version of Wireshark when on Sequoia :()

**Comment by thediveo** [comment]
There is only 4.4.0 at this time and the referenced issue 20069 has no reaction from Wireshark. 20054 doesn't solve the maintenance problems with upgrading and releasing, testing, and installing multiple different packages just because of 20054.

**Comment by thediveo** [comment]
Please make yourself heard by Wireshark, https://gitlab.com/wireshark/wireshark/-/issues/20069 might not happen otherwise. I don't have the resource to clean up things others break and don't own.

**Comment by thediveo** [comment]
Wireshark upstream has fixed two regressions and backported to 4.4.1, scheduled for release on Oct 9th. The existing 0.10.7 extcap integration then will start working again after updating Wireshark to 4.4.1.
